chain/semver_test.go

```diff
@@ -0,0 +1,64 @@
+package chain
+
+import (
+	"testing"
+)
+
+func TestSemverCompatible(t *testing.T) {
+	testCases := []struct {
+		name     string
+		required semver
+		actual   semver
+		expected bool
+	}{
+		{
+			name:     "Equal Versions",
+			required: semver{Major: 1, Minor: 2, Patch: 3},
+			actual:   semver{Major: 1, Minor: 2, Patch: 3},
+			expected: true,
+		},
+		{
+			name:     "Different versions",
+			required: semver{Major: 1, Minor: 2, Patch: 3},
+			actual:   semver{Major: 2, Minor: 0, Patch: 0},
+			expected: false,
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			result := semverCompatible(tc.required, tc.actual)
+			if result != tc.expected {
+				t.Errorf("Expected: %v, got: %v", tc.expected, result)
+			}
+		})
+	}
+	t.Log("Test Passed")
```

Comment:
No need for this. If the test fails, then the standard go tooling will let you know.